src/main/java/com/st/utopia/agent/controller/SeatTickets.java

```diff
@@ -0,0 +1,189 @@
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seats")
+	public ResponseEntity<SeatLocation> getAllSeatsWithPlan(
+			@PathVariable final String flightId) {
+		// FIXME: Search service doesn't yet provide this
+		String url = searchAPI + "/seats?flight=" + flightId;
+		return this.<SeatLocation>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seat/{row}/{seatId}")
+	public ResponseEntity<Ticket> getTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId) {
+		String url = bookingAPI + "/details/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+		return this.<Ticket>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Pay the price for ticket or extend the time
+	 * 
+	 * @return a ticket or an object
+	 */
+	@PutMapping("/flight/{flightId}/seat/{row}/{seatId}/ticket")
+	public ResponseEntity<Object> updateTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final PaymentAmount pay) {
+		// may need to just create a new object
+		if (pay.getPrice() == null) {
+			HttpEntity<PaymentAmount> body = new HttpEntity<>(pay);
+			String url = bookingAPI + "/pay/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+			return this.<Object, PaymentAmount>methodCall(url, HttpMethod.PUT, body);
+		} else {
+			String url = bookingAPI + "/extend/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+			return this.<Object>methodCall(url, HttpMethod.PUT);
+		}
+	}
+
+	/**
+	 * Book flight
+	 * 
+	 * @return a Ticket
+	 */
+	@PostMapping("/flight/{flightId}/seat/{row}/{seatId}/ticket")
+	public ResponseEntity<Ticket> postTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final User reserver) {
+		// not sure if the request body int worked
+		// it should through an error if they do not give a reserver (right?)
+		HttpEntity<User> body = new HttpEntity<>(reserver);
+		String url = bookingAPI + "/book/flights/" + flightId + "/rows/" +
+			row + "/seats/" + seatId;
+		return this.<Ticket, User>methodCall(url, HttpMethod.POST, body);
+	}
+
+	/**
+	 * Cancel a flight if it exists
+	 * 
+	 * @return a Ticket
+	 */
+	@DeleteMapping("/flight/{flightId}/seat/{row}/{seatId}/ticket")
+	public ResponseEntity<Ticket> deleteTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId) {
+		String url = cancellationAPI + "/ticket/flight/" + flightId + "/row/" +
+				row + "/seat/" + seatId;
+		return this.<Ticket>methodCall(url, HttpMethod.PUT);
+	}
+
+	/**
+	 * Get list of tickets for a given bookingCode
+	 * 
+	 * @return list of tickets
+	 */
+	@GetMapping("/booking/{bookingCode}")
+	public ResponseEntity<Ticket> getAllTicketsWithBookingCode(
```

Comment:
Surely this should be just `getTicketWithBookingCode`? (There should be at most one ...)